### PR TITLE
added G502 Lightspeed and Powerplay

### PR DESCRIPTION
**Pull request by F-Lehmann**

### CODE REVIEW

File: RGB.NET.Devices.Logitech/HID/DeviceChecker.cs [hunk 1]
```diff
@@ -53,6 +54,7 @@ internal static class DeviceChecker
                   ("G903", RGBDeviceType.Mouse, 0xC086, 2, "default", @"Mice\G903"),
                   ("G900", RGBDeviceType.Mouse, 0xC539, 2, "default", @"Mice\G900"),
                   ("G703", RGBDeviceType.Mouse, 0xC087, 2, "default", @"Mice\G703"),
+                  ("G502 Lightspeed", RGBDeviceType.Mouse, 0xC231, 2, "default", @"Mice\G502"),
```
Comment (Legion2) [inline RGB.NET.Devices.Logitech/HID/DeviceChecker.cs:57]: C231 is the PID of the Logitech Virtual Mouse

Comment (Legion2) [reply RGB.NET.Devices.Logitech/HID/DeviceChecker.cs:57]: @F-Lehmann can you check if `C539` is the PID of the G502 Lightspeed #126

Comment (F-Lehmann) [reply RGB.NET.Devices.Logitech/HID/DeviceChecker.cs:57]: thats atleast what i found workin gof rme, but i'll have a look

Comment (F-Lehmann) [reply RGB.NET.Devices.Logitech/HID/DeviceChecker.cs:57]: damn, broke everything trying to debug again. only getting RGB.NET.Core.RGBDeviceException in all device providers now.

Comment (F-Lehmann) [reply RGB.NET.Devices.Logitech/HID/DeviceChecker.cs:57]: After some Testing i found out:
1. You are indeed correct C539 is the id for the original receiver of the G502 Lightspeed.
2. If one is using the Powerplay builtin receiver the Virtual mouse with id C231 has to be used.

Comment (Legion2) [reply RGB.NET.Devices.Logitech/HID/DeviceChecker.cs:57]: So the Powerplay can also be the wireless receiver for the G502 Lightspeed (and probably other wireless devices)? Does this mean you have to use different PID in RGB.NET depending on how the Mouse is connected to the PC (original receiver or Powerplay)? Or does the Virtual Mouse PID work for both?
@DarthAffe ok as you already mentioned in #126 multiple devices use the same wireless dongle. I think we should rename the `C539` to something like "wireless mouse receiver" and add the virtual devices.

Comment (F-Lehmann) [reply RGB.NET.Devices.Logitech/HID/DeviceChecker.cs:57]: > So the Powerplay can also be the wireless receiver for the G502 Lightspeed (and probably other wireless devices)?
Yes!
> Does this mean you have to use different PID in RGB.NET depending on how the Mouse is connected to the PC (original receiver or Powerplay)? Or does the Virtual Mouse PID work for both?
The Virtual Mouse seems to always work independent of connection. But i can also use the PID of the connected receiver.
> I think we should rename the C539 to something like "wireless mouse receiver" and add the virtual devices.
This should fix all issues

Comment (F-Lehmann) [reply RGB.NET.Devices.Logitech/HID/DeviceChecker.cs:57]: "So the Powerplay can also be the wireless receiver for the G502 Lightspeed (and probably other wireless devices)?"
Yes!
"Does this mean you have to use different PID in RGB.NET depending on how the Mouse is connected to the PC (original receiver or Powerplay)? Or does the Virtual Mouse PID work for both?"
The Virtual Mouse seems to always work independent of connection. But i can also use the PID of the connected receiver.
"I think we should rename the C539 to something like "wireless mouse receiver" and add the virtual devices."
This should fix all issues

Comment (DarthAffe) [reply RGB.NET.Devices.Logitech/HID/DeviceChecker.cs:57]: `I think we should rename the C539 to something like "wireless mouse receiver" and add the virtual devices.`
I don't get what "add virtual devices" means - The main problem with the logitech sdk is, that it doesn't has any sort of device recognition. It only allows to control "per-key", "per device" or "zone"-devices. That's why we need to estimate what kind of device(s) is/are connected. The exact detection we do is primarily for layouts and the amount of zones (which can vary).

Comment (Legion2) [reply RGB.NET.Devices.Logitech/HID/DeviceChecker.cs:57]: What if we define the generic mice with more zones than the actual? As long as we can control all the lights the users can handle the non functional additional lighting zones which don't exist for their devices.

**Pull request closed without merging**